src/cert.rs

```diff
@@ -110,16 +110,25 @@ impl<'a> Cert<'a> {
                         der::Tag::ContextSpecificConstructed3,
                         Error::TrailingData(DerTypeId::CertificateExtensions),
                         |tagged| {
-                            der::nested_of_mut(
+                            der::nested(
                                 tagged,
                                 der::Tag::Sequence,
-                                der::Tag::Sequence,
                                 Error::TrailingData(DerTypeId::Extension),
-                                |extension| {
-                                    remember_cert_extension(
-                                        &mut cert,
-                                        &Extension::from_der(extension)?,
-                                    )
+                                |outer| {
+                                    while !outer.at_end() {
```

Comment:
I don't think just inlining this is a good solution either given the duplication and lack of clarity. Instead, maybe we can pass an extra `bool` argument to `remember_cert_extension()`?

Reply:
`remember_cert_extension()` is not even called though? The inner `nested` fails when trying to parse a sequence from an empty slice.